src/components/NavBar/MenuItemMobile/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { addLayer, removeLayer } from '../../../context/mapStateSlice';
+import { loadTable } from '../../../context/tableStateSlice';
+import { layersSelector } from '../../../context/mapStateSlice/selectors';
+import { menuList } from '../utils';
+
+function MenuItemMobile({ classes }: MenuItemMobileProps) {
+  const [expanded, setExpanded] = useState<string>('');
+
+  const selectedLayers = useSelector(layersSelector);
+
+  const dispatch = useDispatch();
+
+  const handleChange = (panel: string) => (
+    event: object,
+    newExpanded: boolean,
+  ) => {
+    setExpanded(newExpanded ? panel : '');
+  };
+
+  const toggleLayerValue = (prevChecked: boolean, layer: LayerType) => {
+    if (prevChecked) {
+      dispatch(removeLayer(layer));
+    } else {
+      dispatch(addLayer(layer));
+    }
+  };
+
+  const showTableClicked = (table: TableType) => {
+    dispatch(loadTable(table.id));
+  };
+
+  return (
+    <>
+      {menuList.map(({ title, icon, layersCategories }) => (
+        <Accordion
+          className={classes.accordion}
+          key={title}
+          square
+          expanded={expanded === title}
+          onChange={handleChange(title)}
+        >
+          <AccordionSummary
+            className={classes.accordionSummary}
+            expandIcon={<FontAwesomeIcon icon={faCaretDown} />}
+            IconButtonProps={{ color: 'inherit', size: 'small' }}
+            aria-controls={title}
+            id={title}
+          >
+            <img className={classes.icon} src={icon} alt={title} />
+            <Typography variant="body2">{title}</Typography>
+          </AccordionSummary>
+          <AccordionDetails className={classes.accordionDetail}>
+            <Grid container direction="column">
+              {layersCategories.map(
+                ({ title: categoryTitle, layers, tables }) => (
+                  <Grid
+                    item
+                    key={categoryTitle}
+                    className={classes.categoryContainer}
+                  >
+                    <Typography
+                      variant="body2"
+                      className={classes.categoryTitle}
+                    >
+                      {categoryTitle}
+                    </Typography>
+                    <hr />
+
+                    {layers.map(layer => {
+                      const { id: layerId, title: layerTitle } = layer;
+                      const selected = Boolean(
+                        selectedLayers.find(
+                          ({ id: testId }) => testId === layerId,
+                        ),
+                      );
+                      return (
+                        <div key={layerId} className={classes.layersContainer}>
+                          <Switch
+                            size="small"
+                            color="default"
+                            checked={selected}
+                            onChange={() => toggleLayerValue(selected, layer)}
```

Comment:
It probably isn't too important, but it would be better to fetch the new switch state from the event object from `onChange` (by convention), kind of like how we do so for text fields. This isn't that big of a deal however.

This could be done by making `toggleLayerValue` return a function:
```js
const toggleLayerValue=(selected, layer)=>(event)=>{}
```
where the checked state is stored in: `event.target.checked`